build.gradle

```diff
@@ -1,55 +1,24 @@
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
-buildscript {
-  repositories {
-    maven {
-      url "https://maven.google.com"
-      content {
-        includeGroupByRegex "com\\.android.*"
-        includeGroupByRegex "androidx.*"
-      }
-    }
-    maven {
-      url "https://repo.maven.apache.org/maven2/"
-      content {
-        includeGroupByRegex "org\\.jetbrains.*"
-      }
-    }
-    mavenCentral()
-    jcenter()
-    maven {
-      // For maven publish plugin until 0.9.0 is released
-      url "https://oss.sonatype.org/content/repositories/snapshots/"
-    }
-  }
-  dependencies {
-    classpath 'com.android.tools.build:gradle:3.3.2'
-    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'
-    classpath 'org.jetbrains.dokka:dokka-android-gradle-plugin:0.9.18'
-    classpath 'com.vanniktech:gradle-maven-publish-plugin:0.9.0-SNAPSHOT'
-  }
+plugins {
+  alias(libs.plugins.kotlin.jvm) apply false
+  alias(libs.plugins.kotlin.android) apply false
+  alias(libs.plugins.agp.application) apply false
+  alias(libs.plugins.agp.library) apply false
+  alias(libs.plugins.dokka) apply false
+  alias(libs.plugins.mavenPublish) apply false
 }
 
 allprojects {
-  repositories {
-    maven {
-      url "https://maven.google.com"
-      content {
-        includeGroupByRegex "com\\.android.*"
-        includeGroupByRegex "androidx.*"
+  pluginManager.withPlugin("java") {
+    java {
+      toolchain {
+        languageVersion.set(
+            JavaLanguageVersion.of(libs.versions.jdk.get().removeSuffix("-ea").toInt())
+        )
       }
     }
-    maven {
-      url "https://repo.maven.apache.org/maven2/"
-      content {
-        includeGroupByRegex "org\\.jetbrains.*"
-        includeGroupByRegex "com\\.jakewharton.*"
-        includeGroupByRegex "com\\.squareup.*"
-        includeGroupByRegex "junit.*"
-        includeGroupByRegex "org\\.assertj.*"
-        includeGroupByRegex "org\\.mockito.*"
-      }
+
+    tasks.withType(JavaCompile) {
+      options.release = 8
```

Comment:
Do we want to target 11? As I thought that was the default of AGP 7.4?

Reply:
8 is still the default, 11 (and soon 17) are the requirement for the plugin, but not the compiled sources themselves